### ISSUE TITLE
Failed to enable WebUI "normal" mode, script files are missing!

### ISSUE THREAD

**Issue by gpascualg** [linuxserver/docker-deluge#138]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior

Starting docker should get the webui running and working. Accessing it, after inputting the password, should connect to the daemon, either automatically or manually through the connection manager menu.

## Current Behavior

You can normally access the website, but it fails to connect to the daemon, either manually or automatically.
The logs show the following errors:

```
18:52:14 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
18:52:14 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.
```

You can find the logs in your smoketest servers, which look exactly like mine. Everything, including the website loading fine, is reproduced:

https://ci-tests.linuxserver.io/linuxserver/deluge/latest/index.html

## Steps to Reproduce

1.  Run in docker

```
docker run -d \
  --name=deluge \
  -e PUID=1000 \
  -e PGID=1000 \
  -e TZ=Europe/London \
  -e DELUGE_LOGLEVEL=error `#optional` \
  -p 8112:8112 \
  -p 6881:6881 \
  -p 6881:6881/udp \
  -v /path/to/deluge/config:/config \
  -v /path/to/your/downloads:/downloads \
  --restart unless-stopped \
  lscr.io/linuxserver/deluge
```
2. Access webui, ie, "deluge:8112"
3. Try to connect to the daemon

## Environment
**OS:** Ubuntu 19.04
**CPU architecture:** x86_64
**How docker service was installed:** Official docker repo

## Command used to create docker container (run/create/compose/screenshot)

```
---
version: "2.1"
services:
  deluge:
    image: lscr.io/linuxserver/deluge
    container_name: deluge
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
    volumes:
      - /path/to/deluge/config:/config
      - /path/to/your/downloads:/downloads
    ports:
      - 8112:8112
      - 6881:6881
      - 6881:6881/udp
    restart: unless-stopped
```

## Docker logs

Stripped down from my log to avoid showing any torrents:

```
deluge                     | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
deluge                     | [s6-init] ensuring user provided files have correct perms...exited 0.
deluge                     | [fix-attrs.d] applying ownership & permissions fixes...
deluge                     | [fix-attrs.d] done.
deluge                     | [cont-init.d] executing container initialization scripts...
deluge                     | [cont-init.d] 01-envfile: executing...
deluge                     | [cont-init.d] 01-envfile: exited 0.
deluge                     | [cont-init.d] 01-migrations: executing...
deluge                     | [migrations] started
deluge                     | [migrations] no migrations found
deluge                     | [cont-init.d] 01-migrations: exited 0.
deluge                     | [cont-init.d] 02-tamper-check: executing...
deluge                     | [cont-init.d] 02-tamper-check: exited 0.
deluge                     | [cont-init.d] 10-adduser: executing...
deluge                     |
deluge                     | -------------------------------------
deluge                     |           _         ()
deluge                     |          | |  ___   _    __
deluge                     |          | | / __| | |  /  \
deluge                     |          | | \__ \ | | | () |
deluge                     |          |_| |___/ |_|  \__/
deluge                     |
deluge                     |
deluge                     | Brought to you by linuxserver.io
deluge                     | -------------------------------------
deluge                     |
deluge                     | To support LSIO projects visit:
deluge                     | https://www.linuxserver.io/donate/
deluge                     | -------------------------------------
deluge                     | GID/UID
deluge                     | -------------------------------------
deluge                     |
deluge                     | User uid:    1000
deluge                     | User gid:    1000
deluge                     | -------------------------------------
deluge                     |
deluge                     | [cont-init.d] 10-adduser: exited 0.
deluge                     | [cont-init.d] 30-config: executing...
deluge                     | [cont-init.d] 30-config: exited 0.
deluge                     | [cont-init.d] 90-custom-folders: executing...
deluge                     | [cont-init.d] 90-custom-folders: exited 0.
deluge                     | [cont-init.d] 99-custom-files: executing...
deluge                     | [custom-init] no custom files found exiting...
deluge                     | [cont-init.d] 99-custom-files: exited 0.
deluge                     | [cont-init.d] done.
deluge                     | [services.d] starting services
deluge                     | [services.d] done.
deluge                     | 11:51:29 [INFO    ][deluge.configmanager:52  ] Setting config directory to: /config
deluge                     | 11:51:31 [INFO    ][deluge.core.daemon            :94  ] Deluge daemon 2.0.5
deluge                     | 11:51:31 [INFO    ][deluge.core.core              :339 ] Successfully loaded session.state: /config/session.state
deluge                     | 11:51:31 [INFO    ][deluge.core.core              :339 ] Successfully loaded session.state: /config/session.state.bak
deluge                     | 11:51:31 [INFO    ][deluge.core.alertmanager      :148 ] Alert Queue Size set to 10000
deluge                     | 11:51:31 [INFO    ][deluge.core.rpcserver         :402 ] Starting DelugeRPC server localhost:58846
deluge                     | 11:51:31 [INFO    ][deluge.core.daemon            :154 ] Deluge daemon starting...
deluge                     | 11:51:31 [INFO    ][deluge_label.core             :78  ] *** Start Label plugin ***
deluge                     | 11:51:31 [INFO    ][deluge.pluginmanagerbase      :191 ] Plugin Label enabled...
deluge                     | 11:51:31 [INFO    ][deluge.pluginmanagerbase      :191 ] Plugin ltConfig enabled...
deluge                     | 11:51:31 [INFO    ][deluge.core.authmanager       :228 ] Opening auth for load: /config/auth
deluge                     | 11:51:31 [INFO    ][deluge.core.authmanager       :236 ] Successfully loaded auth: /config/auth
deluge                     | 11:51:31 [INFO    ][deluge.core.torrentmanager    :818 ] Loading torrent state: /config/state/torrents.state
deluge                     | 11:51:31 [INFO    ][deluge.core.torrentmanager    :834 ] Successfully loaded /config/state/torrents.state
deluge                     | 11:51:31 [INFO    ][deluge.core.torrentmanager    :1081] Opening torrents.fastresume for load: /config/state/torrents.fastresume
deluge                     | 11:51:31 [INFO    ][deluge.core.torrentmanager    :1092] Successfully loaded torrents.fastresume: /config/state/torrents.fastresume

*** A few loads ***


deluge                     | 11:51:59 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
deluge                     | 11:51:59 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.

*** More loads ***


deluge                     | 11:52:30 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
deluge                     | 11:52:30 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.

```

**Comment by jseiser** [comment]
Im seeing the same thing


```
15:20:47 [INFO    ][deluge.configmanager:52  ] Setting config directory to: /config
15:20:47 [INFO    ][deluge.core.daemon            :94  ] Deluge daemon 2.0.5
15:20:47 [INFO    ][deluge.core.core              :339 ] Successfully loaded session.state: /config/session.state
15:20:47 [INFO    ][deluge.core.core              :339 ] Successfully loaded session.state: /config/session.state.bak
15:20:47 [INFO    ][deluge.core.alertmanager      :148 ] Alert Queue Size set to 10000
15:20:47 [INFO    ][deluge.core.rpcserver         :402 ] Starting DelugeRPC server :58846
15:20:47 [INFO    ][deluge.core.daemon            :154 ] Deluge daemon starting...
15:20:47 [WARNING ][deluge.core.preferencesmanager:473 ] Unable to find GeoIP database file: /usr/share/GeoIP/GeoIP.dat
15:20:47 [INFO    ][deluge_label.core             :78  ] *** Start Label plugin ***
15:20:47 [INFO    ][deluge.pluginmanagerbase      :191 ] Plugin Label enabled...
15:20:47 [ERROR   ][deluge.pluginmanagerbase      :155 ] Unable to instantiate plugin 'SimpleExtractor' from '/config/plugins/SimpleExtractor-0.4.1-py2.7.egg'!
15:20:47 [ERROR   ][deluge.pluginmanagerbase      :1481] No module named 'core'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/deluge/pluginmanagerbase.py", line 150, in enable_plugin
    instance = cls(plugin_name.replace('-', '_'))
  File "/config/plugins/SimpleExtractor-0.4.1-py2.7.egg/simpleextractor/__init__.py", line 45, in __init__
    from core import Core as _plugin_cls
ModuleNotFoundError: No module named 'core'
15:20:47 [INFO    ][deluge.core.authmanager       :228 ] Opening auth for load: /config/auth
15:20:47 [INFO    ][deluge.core.authmanager       :236 ] Successfully loaded auth: /config/auth
15:20:47 [INFO    ][deluge.core.torrentmanager    :818 ] Loading torrent state: /config/state/torrents.state
15:20:47 [INFO    ][deluge.core.torrentmanager    :834 ] Successfully loaded /config/state/torrents.state
15:20:47 [INFO    ][deluge.core.torrentmanager    :1081] Opening torrents.fastresume for load: /config/state/torrents.fastresume
15:20:47 [INFO    ][deluge.core.torrentmanager    :1092] Successfully loaded torrents.fastresume: /config/state/torrents.fastresume
15:20:47 [INFO    ][deluge.core.torrentmanager    :680 ] Torrent Way of Kings_ Book One of the Stormlight Archive, The - Brandon Sanderson.mobi from user "localclient" loaded
15:20:47 [INFO    ][deluge.core.torrentmanager    :680 ] Torrent Isaac Asimov from user "localclient" loaded
15:20:47 [INFO    ][deluge.core.torrentmanager    :680 ] Torrent South.Park.S25E02.1080p.WEB.h264-BAE from user "localclient" loaded
15:20:47 [INFO    ][deluge.core.torrentmanager    :897 ] Finished loading 3 torrents in 0:00:00.302927
15:20:48 [INFO    ][deluge.core.torrentmanager    :1630] on_alert_external_ip: 198.101.62.6
19:44:29 [INFO    ][deluge.core.rpcserver         :171 ] Deluge Client connection made from: 127.0.0.1:33586
15:20:28 [INFO    ][deluge.core.daemon            :176 ] Deluge daemon shutting down, waiting for components to shutdown...
15:20:28 [INFO    ][deluge.core.core              :307 ] Saving the session.state at: /config/session.state
15:20:28 [INFO    ][deluge.core.rpcserver         :197 ] Deluge client disconnected: Connection to the other side was lost in a non-clean fashion: Connection lost.
15:20:28 [INFO    ][deluge.pluginmanagerbase      :252 ] Plugin Label disabled...
15:20:28 [INFO    ][deluge.core.daemon            :168 ] Deluge daemon shutdown successfully
15:20:28 [INFO    ][deluge.core.daemon_entry      :137 ] Exiting...
Exiting...
```

**Comment by jseiser** [comment]
Im seeing the same issue

```
15:20:43 [INFO    ][deluge.configmanager:52  ] Setting config directory to: /config
15:20:44 [INFO    ][deluge.core.daemon            :94  ] Deluge daemon 2.0.5
15:20:44 [INFO    ][deluge.core.core              :339 ] Successfully loaded session.state: /config/session.state
15:20:44 [INFO    ][deluge.core.core              :339 ] Successfully loaded session.state: /config/session.state.bak
15:20:44 [INFO    ][deluge.core.alertmanager      :148 ] Alert Queue Size set to 10000
15:20:44 [INFO    ][deluge.core.rpcserver         :402 ] Starting DelugeRPC server :58846
15:20:44 [INFO    ][deluge.core.daemon            :154 ] Deluge daemon starting...
15:20:44 [INFO    ][deluge_label.core             :78  ] *** Start Label plugin ***
15:20:44 [INFO    ][deluge.pluginmanagerbase      :191 ] Plugin Label enabled...
15:20:44 [ERROR   ][deluge.pluginmanagerbase      :155 ] Unable to instantiate plugin 'SimpleExtractor' from '/config/plugins/SimpleExtractor-0.4.1-py2.7.egg'!
15:20:44 [ERROR   ][deluge.pluginmanagerbase      :1512] No module named 'core'
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/deluge/pluginmanagerbase.py", line 150, in enable_plugin
    instance = cls(plugin_name.replace('-', '_'))
  File "/config/plugins/SimpleExtractor-0.4.1-py2.7.egg/simpleextractor/__init__.py", line 45, in __init__
    from core import Core as _plugin_cls
ModuleNotFoundError: No module named 'core'
15:20:44 [INFO    ][deluge.core.authmanager       :228 ] Opening auth for load: /config/auth
15:20:44 [INFO    ][deluge.core.authmanager       :236 ] Successfully loaded auth: /config/auth
15:20:44 [INFO    ][deluge.core.torrentmanager    :818 ] Loading torrent state: /config/state/torrents.state
15:20:44 [INFO    ][deluge.core.torrentmanager    :834 ] Successfully loaded /config/state/torrents.state
15:20:44 [INFO    ][deluge.core.torrentmanager    :1081] Opening torrents.fastresume for load: /config/state/torrents.fastresume
15:20:44 [INFO    ][deluge.core.torrentmanager    :1092] Successfully loaded torrents.fastresume: /config/state/torrents.fastresume
15:20:44 [INFO    ][deluge.core.torrentmanager    :680 ] Torrent Way of Kings_ Book One of the Stormlight Archive, The - Brandon Sanderson.mobi from user "localclient" loaded
15:20:44 [INFO    ][deluge.core.torrentmanager    :680 ] Torrent Isaac Asimov from user "localclient" loaded
15:20:44 [INFO    ][deluge.core.torrentmanager    :897 ] Finished loading 2 torrents in 0:00:00.307479
15:20:45 [INFO    ][deluge.core.torrentmanager    :1630] on_alert_external_ip: 198.101.62.6
19:53:22 [INFO    ][deluge.core.rpcserver         :171 ] Deluge Client connection made from: 127.0.0.1:33854
15:07:38 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
15:07:38 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.
15:08:31 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
15:08:31 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.
15:10:07 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
15:10:07 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.
```

**Comment by gpascualg** [comment]
A brief update: somepoint between the issue and today, I could connect to the daemon without issues. I don't now what exactly happened, the warning is still there, but the daemon seems to be working fine.

It seems that after the rebase it takes longer to load all torrents, and the daemon is unresponsive during that time.

**Comment by thelamer** [comment]
That message is normal and always occurred. (regarding the webUI)

**Comment by jeffreyswiggins** [comment]
So you can close this stating this message is expected if you want to but I have run 2 instances of this Deluge Container for over a year and never seen these messages until now and nothing I do and no amount of waiting lets the WebGUI work.  The daemon works so I can connect using the Thick Client but the WebGUI (password enabled) asks for the password and when typed in it just flashes back to asking for it again and you get a lost connection in the log

today at 1:57:52 AM01:57:52 [WARNING ][deluge.ui.web.server        :614 ] Failed to enable WebUI "normal" mode, script files are missing!
today at 1:57:52 AM01:57:52 [WARNING ][deluge.ui.web.server        :626 ] WebUI falling back to "debug" mode.
today at 1:57:52 AM01:57:52 [INFO    ][deluge.core.rpcserver         :197 ] Deluge client disconnected: Connection to the other side was lost in a non-clean fashion: Connection lost.
today at 1:58:23 AM01:58:23 [INFO    ][deluge.core.rpcserver         :171 ] Deluge Client connection made from: 172.22.0.1:44218
today at 1:58:23 AM01:58:23 [INFO    ][deluge.core.rpcserver         :197 ] Deluge client disconnected: Connection to the other side was lost in a non-clean fashion: Connection lost.
today at 1:58:31 AM01:58:31 [INFO    ][deluge.core.rpcserver         :171 ] Deluge Client connection made from: 172.22.0.1:44380
today at 1:58:52 AM01:58:52 [WARNING ][deluge.ui.web.server        :614 ] Failed to enable WebUI "normal" mode, script files are missing!
today at 1:58:52 AM01:58:52 [WARNING ][deluge.ui.web.server        :626 ] WebUI falling back to "debug" mode.

I run 1 container with a dependent VPN connection and the other just as is and both have this same issue.  Started in the last 24 hours for me as I was using the WebGUI less than 12 hours ago and now it will not connect from any web browser it just prompts for the password.  I have taken down the containers hard and had them rebuild and get the same results and also moved to the previous release version from 11 days ago to test and I get the same error messages and same results

**Comment by jeffreyswiggins** [comment]
Also I connected via the Thick Client and attempted to change the password for the account I created to allow access to the WebGUI and I get this **'NoneType' object is not scriptable** error

**Comment by johnnyasantoss** [comment]
Removing `config/auth` and `config/hostlist.conf` fixed the issue for me

**Comment by lilithwhite** [comment]
This issue happened after the last update for me. Nothing like it before and after the update I can't login and get this error:
20:02:41 [WARNING ][deluge.ui.web.server       :614 ] Failed to enable WebUI "normal" mode, script files are missing!
20:02:41 [WARNING ][deluge.ui.web.server       :626 ] WebUI falling back to "debug" mode.
Inserting the correct pw does nothing and a wrong one just says incorrect.

Waiting does not help as I noticed this days after the update and deleting the 2 files mentioned earlier didn't help either. This should really not be closed.

**Comment by laroyru** [comment]
Same here!

> This issue happened after the last update for me. Nothing like it before and after the update I can't login and get this error: 20:02:41 [WARNING ][deluge.ui.web.server :614 ] Failed to enable WebUI "normal" mode, script files are missing! 20:02:41 [WARNING ][deluge.ui.web.server :626 ] WebUI falling back to "debug" mode. Inserting the correct pw does nothing and a wrong one just says incorrect.
> 
> Waiting does not help as I noticed this days after the update and deleting the 2 files mentioned earlier didn't help either. This should really not be closed.

**Comment by ngosang** [comment]
This is a real issue and it's caused because there are some missing files in the Alpine package.

The files used in the Deluge Web UI in "normal mode" are these => https://github.com/deluge-torrent/deluge/blob/ec0bcc11f510cd113bbdb6cd551b27c697f03f53/deluge/ui/web/server.py#L513

But the package only contains:
```
ls -la /usr/lib/python3.10/site-packages/deluge/ui/web/js/
-rw-r--r-- 1 root root 308438 Aug  8 01:37 deluge-all-debug.js
-rw-r--r-- 1 root root 1369003 Aug  8 01:37 extjs/ext-all-debug.js
-rw-r--r-- 1 root root  720033 Aug  8 01:37 extjs/ext-all.js
-rw-r--r-- 1 root root  113353 Aug  8 01:37 extjs/ext-base-debug.js
-rw-r--r-- 1 root root   30625 Aug  8 01:37 extjs/ext-base.js
-rw-r--r-- 1 root root   90943 Aug  8 01:37 extjs/ext-extensions-debug.js
```

if you download the official tarball you can see the file `deluge-all.js` exist but it's not included in Alpine package
![image](https://user-images.githubusercontent.com/10577978/190911629-2a61d8ac-141b-4225-9e7c-7771834658b8.png)

Could you report the issue upstream?

**Comment by necessarily-equal** [comment]
The Alpine Linux package seems to have the 9 js file you are showing: https://pkgs.alpinelinux.org/contents?file=*.js&path=&name=deluge&branch=edge&repo=community&arch=x86_64 (x86_64, package "deluge")

**Comment by necessarily-equal** [comment]
Edit: I got ninja'd on this issue by Alice. https://gitlab.alpinelinux.org/alpine/aports/-/commit/7d4a783d3584e34b775abd3ff8f4148b26756643

I hope this is now fixed then

**Comment by ngosang** [comment]
Thank you both!
It's fixed in `linuxserver/deluge:version-2.1.1-r2`
